webpack.config.js

```diff
@@ -1,62 +1,69 @@
-const path = require('path');
-const ExtractTextPlugin = require('extract-text-webpack-plugin');
-
-const stylesCSSPlugin = new ExtractTextPlugin( {
-	filename: './styles.css',
-} );
+// WordPress dependencies
 
-const adminCSSPlugin = new ExtractTextPlugin( {
-	filename: './admin.css',
-} );
+const defaultConfig = require( '@wordpress/scripts/config/webpack.config.js' );
 
-// Configuration for the ExtractTextPlugin.
-const extractConfig = {
-	use: [
-		{ loader: 'raw-loader' },
-		{
-			loader: 'postcss-loader',
-			options: {
-				plugins: [
-					require( 'autoprefixer' ),
-				],
-			},
-		},
-		{
-			loader: 'sass-loader',
-			query: {
-				outputStyle: 'production' === process.env.NODE_ENV ? 'compressed' : 'nested',
-			},
-		},
-	],
-};
+// External dependencies
+const path = require('path');
+const webpack = require( 'webpack' );
+const MiniCssExtractPlugin = require( 'mini-css-extract-plugin' );
+const FixStyleOnlyEntriesPlugin = require( 'webpack-fix-style-only-entries' );
+const OptimizeCssAssetsPlugin = require( 'optimize-css-assets-webpack-plugin' );
 
 module.exports = {
-	entry: { main: './src/index.js' },
+	...defaultConfig,
+	devtool: 'source-map',
+	externals: {
+		jquery: 'jQuery',
+		$: 'jQuery'
+	},
+	entry: {
+		// JS
+		main: './src/index.js',
+		// SCSS
+		style: './src/styles.scss',
+	},
 	output: {
-		path: path.resolve(__dirname, 'dist'),
-		filename: 'main.js'
+		filename: 'dist/[name].js',
+		path: __dirname
 	},
+	plugins: [
```

Comment:
We should extend `plugins` from the `@wordpress/scripts` configuration so we can utilize the Dependency Extract Plugin: https://github.com/WordPress/gutenberg/blob/master/packages/scripts/README.md#webpack-config

```suggestion
	plugins: [
	  ...defaultConfig.plugins,
```

This will generate a PHP file which we can use to set the dependencies of the script in `wp_enqueue_script`: https://developer.wordpress.org/block-editor/tutorials/block-tutorial/writing-your-first-block-type/#enqueuing-block-scripts

Reply:
We aren't doing this in other products because they support < 5.0 and can't rely on the script dependencies to be available outside of the bundle -- they must be packaged in the bundle in EDD core, EDD Stripe, etc.